client/lib/reader-feed-subscriptions/index.js

```diff
@@ -146,8 +146,7 @@ var FeedSubscriptionStore = {
 		newSubscriptions = Immutable.List( subscriptionsWithState ); // eslint-disable-line new-cap
 
 		// Is it the last page?
-		if ( data.number < perPage ) {
-			// TODO: check to see if this is really the last page based on the count coming from the API
+		if ( data.number === 0 ) {
```

Comment:
OK. So if we had an empty page for some reason... we may still stop early? Probably need to verify that the API is filling in a sane manner.